src/main/webapp/app/entities/course/course-exercises-overview.component.ts

```diff
@@ -1,7 +1,112 @@
 import { Component } from '@angular/core';
+import { Course } from 'app/entities/course/course.model';
+import { CourseExerciseService, CourseService } from 'app/entities/course/course.service';
+import { HttpErrorResponse, HttpResponse } from '@angular/common/http';
+import { QuizExercise } from 'app/entities/quiz-exercise/quiz-exercise.model';
+import { Subscription } from 'rxjs';
+import { ActivatedRoute } from '@angular/router';
+import { QuizExerciseService } from 'app/entities/quiz-exercise/quiz-exercise.service';
+import { JhiAlertService } from 'ng-jhipster';
+import { TextExercise } from 'app/entities/text-exercise/text-exercise.model';
+import { ProgrammingExercise } from 'app/entities/programming-exercise/programming-exercise.model';
+import { ModelingExercise } from 'app/entities/modeling-exercise/modeling-exercise.model';
+import { FileUploadExercise } from 'app/entities/file-upload-exercise/file-upload-exercise.model';
 
 @Component({
     selector: 'jhi-course-overview',
     templateUrl: './course-exercises-overview.component.html'
 })
-export class CourseExercisesOverviewComponent {}
+export class CourseExercisesOverviewComponent {
+    private subscription: Subscription;
+
+    quizExercises: QuizExercise[];
+    textExercises: TextExercise[];
+    programmingExercises: ProgrammingExercise[];
+    modelingExercises: ModelingExercise[];
+    fileUploadExercises: FileUploadExercise[];
+    course: Course;
+    courseId: number;
+
+    constructor(
+        private courseService: CourseService,
+        private route: ActivatedRoute,
+        private jhiAlertService: JhiAlertService,
+        private courseExerciseService: CourseExerciseService,
+        private quizExerciseService: QuizExerciseService
+    ) {}
+
+    ngOnInit() {
+        this.load();
+    }
+
+    load() {
+        this.subscription = this.route.params.subscribe(params => {
+            this.courseId = params['courseId'];
+            this.loadForCourse(this.courseId);
+        });
+    }
+
+    loadForCourse(courseId: number) {
```

Comment:
This is nice.
But where are we using this data?
We are storing it in the `CourseExercisesOverviewComponent` but are not passing it to the exercise components as Inputs in the `course-exercises-overview.component.html` file.

What we are missing is something like this:
```html
<jhi-course-exercise-card headingJhiTranslate="arTeMiSApp.course.programmingExercises">
     <jhi-programming-exercise [showHeading]="false" [course]="course" [exercises]="programmingExercises"></jhi-programming-exercise>
 </jhi-course-exercise-card>
```

At this stage, it would be nice to introduce an abstract component class, which all exercise components inherit from.
This component can declare the `@Input()`s and also define the logic required by these components to only fetch the exercises themselves if they did not get it via the Input.